File: output_20260321_024012/gpt52/java/Linklist/Spouse.java
class Person {

    protected Person spouse;

    /*@
    predicate PersonInv() = this.spouse |-> ?s;
    @*/

    
    public  void spouse_symm()
        
        
    //@ requires PersonInv();
    //@ ensures PersonInv();
    {
        //@ open PersonInv();
        //@ close PersonInv();
        
        
    }

    public Person()
        
        
    //@ requires true;
    //@ ensures PersonInv();
    {
        //@ close PersonInv();
        
    }
    
    public Person getSpouse()
        
        
    //@ requires PersonInv();
    //@ ensures PersonInv() &*& result == spouse;
    {
        //@ open PersonInv();
        Person tmp = spouse;
        //@ close PersonInv();
        return tmp;
        
    }
    
    protected void setSpouse(Person other)
        
        
    //@ requires PersonInv() &*& other.PersonInv();
    //@ ensures PersonInv() &*& other.PersonInv();
    {
        //@ open PersonInv();
        //@ open other.PersonInv();
        spouse = other;
        other.spouse = this;
        //@ close other.PersonInv();
        //@ close PersonInv();
        
    }
    
    protected void clearSpouse()
        
        
    //@ requires PersonInv() &*& spouse != null &*& spouse.PersonInv();
    //@ ensures PersonInv();
    {
        //@ open PersonInv();
        Person s = spouse;
        //@ open s.PersonInv();
        spouse.spouse = null;
        //@ close s.PersonInv();
        spouse = null;
        //@ close PersonInv();
        
    }
    
    void marry(Person other)
        
        
    //@ requires PersonInv() &*& other.PersonInv();
    //@ ensures PersonInv() &*& other.PersonInv();
    {
        other.setSpouse(this);
        
    }
    
    void divorce()
        
        
    //@ requires PersonInv() &*& spouse != null &*& spouse.PersonInv();
    //@ ensures PersonInv() &*& spouse.PersonInv();
    {
        //@ open PersonInv();
        Person s = spouse;
        //@ close PersonInv();
        s.clearSpouse();
        
    }

}

class Program {

    public static void main(String[] args)
        
        
    //@ requires true;
    //@ ensures true;
    {
        Person a = new Person();
        Person b = new Person();
        a.marry(b);
        b.divorce();
    }

}

File: output_repair/claude-opus/sample_1/repair_2/c/c/Struct/spouse.c
#include "stdlib.h"

struct person
{
  char *name;
  struct person *spouse;
};

/*@ predicate person(struct person *p;) = 
      p->name |-> _ &*& p->spouse |-> _; @*/

struct person *create_person()
//@ requires true;
//@ ensures person(result) &*& result->spouse |-> 0;
{
  struct person *p = malloc(sizeof(struct person));
  if (p == 0)
    abort();
  p->spouse = 0;

  return p;
}

void marry(struct person *this, struct person *other)
//@ requires person(this) &*& person(other);
//@ ensures person(this) &*& person(other) &*& this->spouse |-> other &*& other->spouse |-> this;
{
  this->spouse = other;
  other->spouse = this;
}

struct person *person_get_spouse(struct person *this)
//@ requires this->spouse |-> ?s;
//@ ensures this->spouse |-> s &*& result == s;
{
  return this->spouse;
}

void divorce(struct person *this)
//@ requires person(this) &*& this->spouse |-> ?other &*& other != 0 &*& person(other) &*& other->spouse |-> this;
//@ ensures person(this) &*& this->spouse |-> 0 &*& person(other) &*& other->spouse |-> 0;
{
  this->spouse->spouse = 0;
  this->spouse = 0;
}

void die(struct person *this)
//@ requires person(this) &*& this->spouse |-> 0;
//@ ensures true;
{
  if (this->spouse != 0)
  {
    this->spouse->spouse = 0;
  }
  free(this);
}

int main()
//@ requires true;
//@ ensures true;
{
  struct person *alice = create_person();
  struct person *bob = create_person();
  struct person *eve = 0;
  marry(alice, bob);
  eve = create_person();
  divorce(bob);
  marry(bob, eve);
  die(alice);
  die(bob);
  die(eve);
  return 0;
}